CMakeLists.txt

```diff
@@ -13,31 +13,41 @@ endif()
 
 option(REDNER_CUDA "Build redner with GPU code path?" OFF)
 
+# TODO: CUDA does currently not work with Win32 as
+#       pybind11_add_module creates the redner target but the
+#       call to cuda_add_library tries to create it again.
+if(WIN32 AND REDNER_CUDA)
+  set(REDNER_CUDA OFF)
+  message(WARNING "GPU build currently not supported for Win32.")
+endif()
+
 if(REDNER_CUDA)
     message(STATUS "Build with CUDA support")
     find_package(CUDA 10 REQUIRED)
+    # We don't set REQUIRED here as the main OptiX lib will not be found
+    # (only OptiX Prime is required)
+    find_package(OptiX)
 else()
     message(STATUS "Build without CUDA support")
     find_package(Thrust REQUIRED)
 endif()
 
-set(EMBREE_INCLUDE_PATH "${PROJECT_SOURCE_DIR}/redner-dependencies/embree/include")
-if(APPLE)
-    set(EMBREE_LIBRARY "${PROJECT_SOURCE_DIR}/redner-dependencies/embree/lib-macos/libembree3.dylib")
-else()
-    set(EMBREE_LIBRARY "${PROJECT_SOURCE_DIR}/redner-dependencies/embree/lib-linux/libembree3.so")
-endif()
+find_package(Embree REQUIRED)
 
 include_directories(${PYTHON_INCLUDE_PATH})
 include_directories(${EMBREE_INCLUDE_PATH})
 if(REDNER_CUDA)
-    include_directories(${CMAKE_SOURCE_DIR}/redner-dependencies/optix/include)
+    include_directories(${OptiX_INCLUDE})
     link_directories(${CUDA_LIBRARIES})
 else()
     include_directories(${THRUST_INCLUDE_DIR})
 endif()
 
-add_compile_options(-Wall -g -O3 -fvisibility=hidden)
+if(NOT MSVC)
+  # These compile definitions are not meaningful for MSVC
+  add_compile_options(-Wall -g -O3 -fvisibility=hidden)
```

Comment:
Optimizations should still be enabled with MSVC. More warnings and debuginfo would also be useful.

Reply:
I agree with the debug info and warnings but not with the optimizations. If you turn on optimizations, you cannot debug the code line by line anymore as some stuff is optimized away. I wouldn't pre-bake that in the `CMakeLists.txt` but rather let the user decide. Remember in VS you can switch between `Debug` and `Release` within the same solution, where `Release` implies compilation with optimizations.

Reply:
I added the all warnings and debug info flags for MSVC in the latest commit.